sssd_test_framework/utils/sssd.py

```diff
@@ -875,20 +875,25 @@ def autofs(self) -> None:
         self.sssd.authselect.select("sssd")
         self.sssd.enable_responder("autofs")
 
-    def mkhomedir(self, homedir: str = "/home") -> None:
+    def mkhomedir(self, homedir: str = "/home", clean: bool | None = False) -> None:
         """
         Configure SSSD with mkhomedir and oddjobd.
 
         :param homedir: Home directory path.
         :type homedir: str | None, optional
+        :param clean: Clears contents for a clean directory.
+        :type clean: bool | None, optional
 
         #. Select authselect sssd profile with 'with-mkhomedir'
         #. Start oddjobd.service
         #. Backup home directory
+        #. Clean home directory if set to ``True``
         """
         self.sssd.authselect.select("sssd", ["with-mkhomedir"])
         self.sssd.svc.start("oddjobd.service")
         self.sssd.fs.backup(homedir)
+        if clean:
+            self.sssd.fs.rm(f"{homedir}/*")
```

Comment:
Can you check that the wildcard actually works? The command expands to `rm -fr 'homedir/*'` in single quotes, this should block the wildcard.

Reply:
Can you point me to a test that is fixed by this?

Reply:
https://github.com/SSSD/sssd/pull/7682

It's hard to work on because this is failing in PRCI only. I've temporarily added print statements to a commit to see the directory's contents, and it does contain artifacts from existing tests. It can't create the directory because a homedir already exists and isn't owned by the user.

Reply:
I will take some time and look into the issue. First, this line can not work, the wildcard will not be expanded. Second, the homedir is already backed up on the previous line so it should be restored to the original content after each test. If this is not happening, then we have a bug.

Reply:
@pbrezina , right, because the home dir is not empty, backup, clear, and restore to the state before the test is run, which includes some directories. FWIW, I do not see /home on the list to be backed up, so if a test does create a directory, it will persist.  

https://github.com/SSSD/sssd-test-framework/blob/7bcb65e2c47610aafc34d01494820bfe90c7ca22/sssd_test_framework/hosts/client.py#L91

So we add /home to that list, maybe enough to fix it.

Reply:
there is `self.sssd.fs.backup(homedir)` right before your code, which defaults to /home. So /home is backed up once mkhomedir is setup. Noting that is created by it sould persist.

Reply:
https://github.com/SSSD/sssd/actions/runs/11959670217/job/33343281251?pr=7682#step:12:203

I added a 'ls -ltrh /home' before the setup, I did this before and I saw directories created from what I assume are previous tests. The host.backup doesn't restore this folder, so any content that is created from other tests will persist. I also added a clear statement, so they should also pass now. We should be able to look at the runner.log and see the contents before the method is called, to confirm.

Reply:
host.backup does not restore it, fs module takes care of that

Reply:
so the test is now failing for all providers, not just samba and this is because mkhomedir is in system-auth and oddjob is started by default, so other tests where user actually authenticates creates the home dir as a side effect. Adding /home to ClientHost.backup/restore will work.

I would:
* add backup and restore of /home to ClientHost
* ditch the commit that adds clean flag to mkhomedir as the wildcard won't work
* keep the second commit that adds clean to sssd, it probably is not needed in this case (I'd remove it from the test), but it may be helpful in other tests

Reply:
Exactly, so I think the best fix is to update host.backup to backup /home/ so the state is clean over the clear in this PR.

Reply:
I added another commit, add /home to backup and restore in hosts/client.py  , this should keep the directory clean.

Reply:
Ditch the first commit and its an ack.

Reply:
@pbrezina How about actually fixing the wildcard for rm?  This is an unwanted/unexpected limitation that will just push people to use command rm directly instead.